tests/cases/compiler/argumentsSpreadRestIterables.tsx

```diff
@@ -0,0 +1,18 @@
+// @strict: true
+// @noEmit: true
+// @target: es5, esnext
+
+declare const itNum: Iterable<number>
+
+;(function(...rest) {})(...itNum)
+;(function(a, ...rest) {})('', ...itNum)
+;(function(a, ...rest) {})('', true, ...itNum)
+
+declare function fn1<const T extends readonly unknown[]>(...args: T): T;
+const res1 = fn1(..."hello");
```

Comment:
I would also prefer to handle this in a follow-up PR - but one additional issue comes to mind here. Should this produce `readonly string[]` (like now) or should it actually be `readonly ['h', 'e', 'l', 'l', 'o']`? :P

Reply:
It seems complicated to do that, so I'd probably say to start with the easier option.

Reply:
Ye, I definitely want to start with the easier option (what is implemented today). Spreading into characters wouldn't have to be that complicated though since we could just expand the string into an array using JS itself:
```ts
[...'hello'].map(char => /* create string literal types */)
```

This solution has a nice advantage... it would work correctly with emojis 👨‍🎤 

There is one potential problem with this though and I will touch on it in a different comment.